### PR TITLE
add clear user instructions for dataset.prefetch

### PR DESCRIPTION
**Pull request by liutongxuan**
The argument's name "buffer_size" and api_doc "Creates a Dataset that prefetches elements from this dataset." mislead users, which could lead to huge memory usage. Several users made such misunderstanding which trigger huge memory usage, similar confuse in stack overflow: https://stackoverflow.com/questions/49707062/tensorflow-dataset-api-dataset-batchn-prefetchm-prefetches-m-batches-or-sam

### CODE REVIEW

**Review by jsimsa — DISMISSED** [review]

**Comment by rthadur** [comment]
@liutongxuan could you please check build errors

**Comment by liutongxuan** [comment]
> @liutongxuan could you please check build errors

Not related, the PR only contains comments changes.

**Comment by jsimsa** [comment]
The failures are related, in particular https://source.cloud.google.com/results/invocations/1a624d60-a977-4faa-b9b9-a7539aebfad4/log indicates that your comment changes violate the 80 character per line limit:

```
FAIL: Found 2 non-whitelited pylint errors:
tensorflow/python/data/ops/dataset_ops.py:623: [C0301(line-too-long), ] Line too long (82/80)
tensorflow/python/data/ops/dataset_ops.py:624: [C0301(line-too-long), ] Line too long (83/80)
```

File: tensorflow/python/data/ops/dataset_ops.py [hunk 1]
```diff
@@ -648,6 +648,12 @@ def concatenate(self, dataset):
   def prefetch(self, buffer_size):
     """Creates a `Dataset` that prefetches elements from this dataset.
 
+    Note: Like other `Dataset` methods, prefetch operates on the elements of
+    the dataset. It has no concept of examples vs. batches. `examples.prefetch(2)`
```
Comment (jsimsa) [inline tensorflow/python/data/ops/dataset_ops.py]: This and the next line violate the 80 character limit.

Comment (liutongxuan) [reply tensorflow/python/data/ops/dataset_ops.py]: thx, done

**Comment by rthadur** [comment]
@liutongxuan can you please resolve conflicts.

**Comment by liutongxuan** [comment]
> @liutongxuan can you please resolve conflicts.

done, thx

**Comment by liutongxuan** [comment]
@rthadur  fixed conflict, thx

File: tensorflow/python/data/ops/dataset_ops.py [hunk 2]
```diff
@@ -699,8 +699,12 @@ def concatenate(self, dataset):
   def prefetch(self, buffer_size):
     """Creates a `Dataset` that prefetches elements from this dataset.
 
-    Note that if the dataset was batched using `Dataset.batch`, each element is
-    a batch and this operation will prefetch `buffer_size` batches.
+    Note: Like other `Dataset` methods, prefetch operates on the
+    elements of the dataset. It has no concept of examples vs. batches.
```
Comment (jsimsa) [inline tensorflow/python/data/ops/dataset_ops.py]: replace "elements of the dataset" with "elements of thei nput dataset".

File: tensorflow/python/data/ops/dataset_ops.py [hunk 3]
```diff
@@ -699,8 +699,12 @@ def concatenate(self, dataset):
...
+    `examples.prefetch(2)` will prefetch two elements (2 examples),
+    while `examples.batch(20).prefetch(2)` will prefetch 2 elements
+    (2 batches, of 20 examples each). The typical use-case is to
```
Comment (jsimsa) [inline tensorflow/python/data/ops/dataset_ops.py]: Remove the last sentence. It might a typical use case for yourself, but not for others.

**Review by jsimsa — APPROVED** [review]

**Pull request merged**